datafusion/core/src/dataframe/mod.rs

```diff
@@ -1536,10 +1537,20 @@ impl DataFrame {
                 .build()?
         };
 
+        let table_ref: TableReference = table_name.into();
+        let table = table_ref.table().to_string();
+        let table_schema = self.session_state.schema_for_ref(table_ref)?;
+        let target = match table_schema.table(&table).await? {
+            Some(ref provider) => Ok(Arc::clone(provider)),
+            _ => plan_err!("No table named '{table}'"),
+        }?;
+
+        let target = Arc::new(DefaultTableSource::new(target));
```

Comment:
I wasn't sure why it was required to go from `table_name` --> `table_ref` --> `table` (another name) again. I tried this locally and it seemed to work well (all tests passed)

```suggestion
        let table_ref: TableReference = table_name.into();
        let table_schema = self.session_state.schema_for_ref(table_ref.clone())?;
        let target = match table_schema.table(table_ref.table()).await? {
            Some(ref provider) => Ok(Arc::clone(provider)),
            _ => plan_err!("No table named '{table_name}'"),
        }?;

        let target = Arc::new(DefaultTableSource::new(target));
```